strum_macros/src/macros/enum_variant_names.rs

```diff
@@ -24,10 +23,10 @@ pub fn enum_variant_names_inner(ast: &syn::DeriveInput) -> TokenStream {
         .collect::<Vec<_>>();
 
     quote! {
-        impl #name {
+        impl VariantNames for #name {
```

Comment:
We have to do a little work here to match types with generic parameters. See the [EnumIter macro](https://github.com/Peternator7/strum/blob/e981f240d6e02c49f246b0fc4c7f4978e5ff364b/strum_macros/src/macros/enum_iter.rs#L68) for an example. I suspect the old implementation missed this so it would be great if you could fix it.

Reply:
Also make sure we use an absolute path to the trait to avoid collisions